bigtable-dataflow-parent/bigtable-beam-import/src/main/java/com/google/cloud/bigtable/beam/TemplateUtils.java

```diff
@@ -45,6 +47,14 @@ public static CloudBigtableTableConfiguration BuildImportConfig(ImportOptions op
     if (opts.getBigtableAppProfileId() != null) {
       builder.withAppProfileId(opts.getBigtableAppProfileId());
     }
+
+    ValueProvider enableThrottling = ValueProvider.NestedValueProvider.of(
+        opts.getBigtableWriteThrottleMs(), (Integer throttleMs) -> String.valueOf(throttleMs > 0));
+
+    builder.withConfiguration(BigtableOptionsFactory.BIGTABLE_BUFFERED_MUTATOR_ENABLE_THROTTLING, enableThrottling);
+    builder.withConfiguration(BigtableOptionsFactory.BIGTABLE_BUFFERED_MUTATOR_THROTTLING_THRESHOLD_MILLIS,
+        String.valueOf(opts.getBigtableWriteThrottleMs()));
```

Comment:
I don't think this is right: `opts.getBigtableWriteThrottleMs()` will return a ValueProvider, so I don't think you want to wrap the actual ValueProvider, but instead the value of the ValueProvider. I think you need to use a NestedValueProvider here as well.

``` java
NestedValueProvider.of(opts.getBigtableWriteThrottleMs(), Integer::toString)
```

Reply:
The point is we need to defer reading the value of a runtime parameter at pipeline construction time. This is kind of annoying that we have to convert from non-string type to string type. We should be able to make this simpler.

Reply:
Ah, ok. Understood.  I didn't realize that subtlety.